overwatch/src/services/runner.rs

```diff
@@ -0,0 +1,350 @@
+                    }
+
+                    // TODO: Sending a different signal could be handy to differentiate whether
+                    //  the service was already started or not.
+                    if let Err(error) = finished_signal_sender.send(()) {
+                        dbg!(
+                            "Error while sending the LifecycleMessage::Start signal: {}.",
+                            error
+                        );
+                    }
+                }
+                LifecycleMessage::Stop(finished_signal_sender) => {
+                    if service_lifecycle_phase == LifecyclePhase::Stopped {
+                        info!("Service is already stopped.");
+                    } else {
+                        Self::handle_stop(
+                            &mut service_task_handle,
+                            &mut state_handle_task_handle,
+                            &mut service_resources,
+                        )
+                        .await;
+                        service_lifecycle_phase = LifecyclePhase::Stopped;
+                    }
+
+                    // TODO: Sending a different signal could be handy to differentiate whether
+                    //  the service was already stopped or not.
+                    if let Err(error) = finished_signal_sender.send(()) {
+                        dbg!("Error while sending the LifecycleMessage::Stop finished signal: {}. Likely due to the receiver being already dropped in the Service::run task.", error);
+                    }
+                }
+            }
+        }
+    }
+
+    fn handle_start<Service>(
+        service_resources: &mut ServiceResources<
+            Message,
+            Settings,
+            State,
+            StateOp,
+            RuntimeServiceId,
+        >,
+        service_task_handle: &mut Option<JoinHandle<()>>,
+        state_handle_task_handle: &mut Option<JoinHandle<()>>,
+    ) where
+        Service: ServiceCore<RuntimeServiceId, Settings = Settings, State = State, Message = Message>
+            + 'static,
+        StateOp: Clone,
+    {
+        let initial_state = match service_resources.get_service_initial_state() {
+            Ok(initial_state) => initial_state,
+            Err(error) => {
+                panic!("Failed to create the initial state from settings: {error}");
+            }
+        };
+
+        let inbound_relay = service_resources
+            .inbound_relay
+            .take()
+            .expect("Failed to retrieve inbound relay.");
+
+        let service_resources_handle = service_resources.to_handle(inbound_relay);
+        let service = Service::init(service_resources_handle, initial_state.clone());
+
+        service_resources
+            .state_updater()
+            .update(Some(initial_state));
+
+        service_resources
+            .status_handle
+            .service_runner_updater()
+            .starting();
+
+        match service {
```

Comment:
Why don't we panic right after we create the service?

Reply:
Good point. Likely a vestige of how it was handled in one of the previous iterations.